contracts/0.4.24/lib/PositiveTokenRebaseLimiter.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.4.24;
+
+import {SafeMath} from "@aragon/os/contracts/lib/math/SafeMath.sol";
+
+import {Math256} from "../../common/lib/Math256.sol";
+
+/**
+ * This library implements positive rebase limiter for `stETH` token.
+ * One needs to initialize `LimiterState` with the desired parameters:
+ * - _rebaseLimit (limiter max value, nominated in LIMITER_PRECISION_POINTS)
+ * - _totalPooledEther (see `Lido.getTotalPooledEther()`)
+ * - _totalShares (see `Lido.getTotalShares()`)
+ *
+ * The limiter allows to account for:
+ * - consensus layer balance updates (can be either positive or negative)
+ * - total pooled ether changes (withdrawing funds from vaults on execution layer)
+ * - total shares changes (coverage application)
+ */
+
+library LimiterState {
+    /**
+      * @dev Internal limiter representation struct (storing in memory)
+      */
+    struct Data {
+        uint256 totalPooledEther;  // total pooled ether pre-rebase
+        uint256 totalShares;       // total shares before pre-rebase
+        uint256 rebaseLimit;       // positive rebase limit (target value)
+        uint256 accumulatedRebase; // accumulated rebase (previous value)
+    }
+}
+
+library PositiveTokenRebaseLimiter {
+    using SafeMath for uint256;
+
+    /// @dev Precision base for the limiter (e.g.: 1e6 - 0.1%; 1e9 - 100%)
+    uint256 private constant LIMITER_PRECISION_BASE = 10**9;
+    /// @dev Disabled limit
+    uint256 private constant UNLIMITED_REBASE = uint256(-1);
+
+    /**
+      * @dev Initialize the new `LimiterState` structure instance
+      * @param _rebaseLimit max limiter value (saturation point), see `LIMITER_PRECISION_POINTS`
+      * @param _totalPooledEther total pooled ether, see `Lido.getTotalPooledEther()`
+      * @param _totalShares total shares, see `Lido.getTotalShares()`
+      * @return newly initialized limiter structure
+      */
+    function initLimiterState(
+        uint256 _rebaseLimit,
+        uint256 _totalPooledEther,
+        uint256 _totalShares
+    ) internal pure returns (LimiterState.Data memory _limiterState) {
+        require(_rebaseLimit > 0, "TOO_LOW_TOKEN_REBASE_MAX");
+        require(_rebaseLimit <= UNLIMITED_REBASE, "WRONG_REBASE_LIMIT");
```

Comment:
is redundant check?

Reply:
yeah, good catch ♥
though, it's refactored in another branch to have `uint64.max` as `UNLIMITED`